langlib/lang.function/src/main/java/org/ballerinalang/langlib/function/Call.java

```diff
@@ -0,0 +1,134 @@
+import io.ballerina.runtime.internal.scheduling.Scheduler;
+import io.ballerina.runtime.internal.types.BArrayType;
+import io.ballerina.runtime.internal.types.BFunctionType;
+import io.ballerina.runtime.internal.types.BTupleType;
+import io.ballerina.runtime.internal.util.exceptions.BLangExceptionHelper;
+import io.ballerina.runtime.internal.util.exceptions.RuntimeErrors;
+import io.ballerina.runtime.internal.values.ArrayValueImpl;
+import io.ballerina.runtime.internal.values.ListInitialValueEntry;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.ballerina.runtime.api.constants.RuntimeConstants.BALLERINA_BUILTIN_PKG_PREFIX;
+import static io.ballerina.runtime.api.constants.RuntimeConstants.FUNCTION_LANG_LIB;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.INCOMPATIBLE_TYPES;
+import static io.ballerina.runtime.internal.util.exceptions.BallerinaErrorReasons.getModulePrefixedReason;
+
+/**
+ * Native implementation of lang.function:call(function func, any|error... args).
+ *
+ * @since 2201.2.0
+ */
+public class Call {
+
+    private static final StrandMetadata METADATA = new StrandMetadata(BALLERINA_BUILTIN_PKG_PREFIX, FUNCTION_LANG_LIB,
+                                                                      "1.0.0", "call");
+
+    public static Object call(BFunctionPointer<Object, Object> func, Object... args) {
+        BFunctionType functionType = (BFunctionType) func.getType();
+        List<Type> paramTypes = new LinkedList<>();
+        List<Type> argTypes = new LinkedList<>();
+        List<Object> argsList = new java.util.ArrayList<>(List.of(Scheduler.getStrand()));
+
+        if (validatePositionalArgs(args, argsList, functionType, paramTypes, argTypes) ||
+                validateRestArgs(args, argsList, functionType, paramTypes, argTypes)) {
+                throw ErrorCreator.createError(
+                        getModulePrefixedReason(FUNCTION_LANG_LIB, INCOMPATIBLE_TYPES),
+                        BLangExceptionHelper.getErrorDetails(RuntimeErrors.INCOMPATIBLE_TYPE,
+                                                             new BTupleType(paramTypes), new BTupleType(argTypes)));
+        }
+
+        return func.asyncCall(argsList.toArray(), METADATA);
+    }
+
+    private static boolean validatePositionalArgs(Object[] args, List<Object> argsList, BFunctionType functionType,
+                                                  List<Type> paramTypes, List<Type> argTypes) {
+        boolean errored = false;
+        Parameter[] parameters = functionType.parameters;
+        int numOfParams = parameters.length;
+        int numOfArgs = args.length;
+        for (int i = 0; i < numOfParams; i++) {
+            Parameter parameter = parameters[i];
+            Type paramType = parameter.type;
+            paramTypes.add(paramType);
+            if (i >= numOfArgs) {
+                if (parameter.isDefault) {
+                    argsList.add(0);
```

Comment:
Does this have to be 0? What do we usually pass for b7a code that's desugared?

Reply:
No, this can be any value.
In desugar, we created `IGNORE_EXPR` for this. When generating Java code we set it as `0`.